kn/base/templatetags/base.py

```diff
@@ -25,6 +26,15 @@ def email_filter(value):
 def mark_safe_filter(value):
     return mark_safe(value)
 
+@stringfilter
+@register.filter(name='json')
+def json_filter(data):
+    return mark_safe(json.dumps(data)
+                         .replace('&', '\u0026')
+                         .replace('<', '\u003c')
+                         .replace('>', '\u003e')
+                         .replace('/', '\u002f'))
```

Comment:
Niet dat ik zo snel heb kunnen vinden.
Het probleem is dat een standaard filter voor HTML ook `"` codeerd, waardoor de JSON niet meer valid is. Voor zover ik weet kun je met alleen `"` en `'` geen XSS aanval doen, als je de JSON niet in een HTML attribuut stopt. Zie [OWASP](https://www.owasp.org/index.php/XSS_%28Cross_Site_Scripting%29_Prevention_Cheat_Sheet#JSON_entity_encoding).
Ik heb de filter gebaseerd op deze tekst:
> An alternative to escaping and unescaping JSON directly in JavaScript, is to normalize JSON server-side by converting '<' to '\u003c' before delivering it to the browser.

Zie ook https://code.djangoproject.com/ticket/17419 (in het kort: te moeilijk dus doen we niet).

Bovendien, dit zijn nog steeds leden die foto's uploaden dus heel groot lijkt me de kans niet dat er misbruik van wordt gemaakt.